Arie comments

yoav-el-certora opened the pull request:
Naming convention:
 - PRs for features that are in design should have the "proposal" label
 - PRs for features that haven't landed yet should have the "future" label
 - PRs for upcoming releases should have the "release" label
 - PRs with new documentation for existing features should have the "existing feature" label

Before requesting review:
 - Make sure there is a ticket in the DOC board in Jira
 - Make sure CI is passing
   - Spell check failure may require adding backticks around code or updating `spelling_wordlist.txt`
   - See `README.md` for information about style and markdown syntax
   - If the CI Details link gives a 404, you need to log in to readthedocs.com
 - Add link to generated documentation here
   - you can find this by following the read the docs link from the CI check
 - Ask for help in #documentation

Jira ticket: TODO
Link to generated documentation: TODO

phreppo commented on docs/solana/options.md:
````diff
@@ -86,13 +92,15 @@ See an [example](./scripts/cvlr_inlining.txt).
+certoraSolanaProver --build_script <path_to_build_script> --solana_inlining <path_to_inlining_file>  --rule rule_some_rule
 ```
 
+(--solana_summaries)=
````
See previous comment

phreppo commented on docs/solana/options.md:
```diff
@@ -104,11 +112,12 @@ See an [example](./scripts/cvlr_summaries.txt).
 **When to use it?**
 
 This option is currently required for every project.
+It can be provided to the prover by passing this files as flag or by retrieving it from the build_script.
```
```suggestion
It can be provided to the Prover by passing this file as a flag or by retrieving it from the build script.
```

phreppo commented on docs/solana/options.md:
````diff
@@ -104,11 +112,12 @@ See an [example](./scripts/cvlr_summaries.txt).
 
 ```bash
-certoraSolanaProver --build_script <path_to_build_script> --solana_summaries <path_to_summaries_file>
+certoraSolanaProver --build_script <path_to_build_script> --solana_summaries <path_to_summaries_file> --rule rule_some_rule
````
rule name

phreppo commented on docs/solana/options.md:
````diff
@@ -125,7 +134,7 @@ Use it when there is a need to enable a specific [Cargo feature](https://doc.rus
 
 ```bash
-certoraSolanaProver --build_script <path_to_build_script> --cargo_features <feature_1> <feature_2>
+certoraSolanaProver --build_script <path_to_build_script> --cargo_features <feature_1> <feature_2> --rule rule_some_rule
````
rule name

phreppo commented on docs/solana/options.md:
````diff
@@ -143,7 +152,7 @@ It is also helpful to keep track of a single file verification status over time,
 
 ```bash
-certoraSolanaProver --build_script <path_to_build_script> --msg 'Removed an assertion'
+certoraSolanaProver --build_script <path_to_build_script> --msg 'Removed an assertion' --rule rule_some_rule
````
rule name

phreppo reviewed (commented):
I think that this is missing: https://github.com/Certora/Documentation/pull/335#discussion_r1927788389.
I did not check, but are the following addressed?
* https://github.com/Certora/Documentation/pull/335#discussion_r1927792249
* https://github.com/Certora/Documentation/pull/335#discussion_r1927794697

Spell checking is failing as well.

yoav-el-certora commented on docs/solana/options.md:
```diff
@@ -37,12 +37,17 @@ These modes are mutually exclusive - you cannot run the tool with more than one
 Specifies the location of the script that has to be called to compile the Rust project.
-The build script must output the following:
+The build script should output 0 on success and 1 on failure unless its being executed using the ``--json`` flag.
+In this case, the build script should output the following:
```
This is printed directly to the stdout, in any case this is a semi-internal feature.

yoav-el-certora commented on docs/solana/options.md:
```diff
@@ -37,12 +37,17 @@ These modes are mutually exclusive - you cannot run the tool with more than one
 - ``success``: Boolean flag indicating if the build phase passed successfully.
+- ``return_code``: The return code of the script.
+- ``log``: Optionally provide paths to files for stdout and stderr of the build.
+- ``solana_inlining``: List of paths to [inlining](#--solana_inlining) files for Solana contracts.
+- ``solana_summaries``: List of paths to [summaries](#--solana_summaries) files for Solana contracts.
```
The feature will land soon.

yoav-el-certora commented on docs/solana/options.md:
```diff
@@ -86,13 +92,15 @@ See an [example](./scripts/cvlr_inlining.txt).
 **When to use it?**
 
 This option is currently required for every project.
+It can be provided to the prover by passing this files as flag or by retrieving it from the build_script.
```
Changed to list, as this is the input

yoav-el-certora commented:
> I think that this is missing: [#335 (comment)](https://github.com/Certora/Documentation/pull/335#discussion_r1927788389). I did not check, but are the following addressed?
> 
> * [DOC-433 | CERT-7992 | Added solana cli options #335 (comment)](https://github.com/Certora/Documentation/pull/335#discussion_r1927792249)
> * [DOC-433 | CERT-7992 | Added solana cli options #335 (comment)](https://github.com/Certora/Documentation/pull/335#discussion_r1927794697)
> * [DOC-433 | CERT-7992 | Added solana cli options #335 (comment)](https://github.com/Certora/Documentation/pull/335#discussion_r1927795728)
> 
> Spell checking is failing as well.

Addressed comments which are relevant. Some fixes you already did in the main branch after merging the original PR.
